docs/Account.md

```diff
@@ -0,0 +1,331 @@
+
+public_key: felt
+```
+
+#### `get_address`
+
+Returns the contract address of the Account.
+
+##### Parameters:
+
+    None
+
+##### Returns:
+
+```
+(implicit) syscall_ptr : felt*
+(implicit) pedersen_ptr: HashBuiltin*
+(implicit) range_check_ptr
+
+contract_address: felt
+```
+
+#### `get_nonce`
+
+Returns the current transaction nonce for the Account.
+
+##### Parameters:
+```
+None
+```
+##### Returns:
+
+```
+(implicit) syscall_ptr : felt*
+(implicit) pedersen_ptr: HashBuiltin*
+(implicit) range_check_ptr
+```
+
+#### `set_public_key`
+
+Sets the public key that will control this Account. It can be used to rotate keys for security, change them in case of compromised keys or even transferring ownership of the account.
+
+##### Parameters:
+```
+public_key: felt
+```
+
+##### Returns:
+```
+(implicit) syscall_ptr : felt*
+(implicit) pedersen_ptr: HashBuiltin*
+(implicit) range_check_ptr
+```
+
+#### `is_valid_signature`
+
+This function is inspired in [EIP-1271](https://eips.ethereum.org/EIPS/eip-1271) and checks whether a given signature is valid, otherwise it reverts.
+
+##### Parameters:
+```
+(implicit) syscall_ptr : felt*
+(implicit) pedersen_ptr : HashBuiltin*
+(implicit) range_check_ptr
+(implicit) ecdsa_ptr: SignatureBuiltin*
+
+hash: felt
+signature_len: felt
+signature: felt*
+```
+
+##### Returns:
+```
+(implicit) syscall_ptr : felt*
+(implicit) pedersen_ptr: HashBuiltin*
+(implicit) range_check_ptr
+(implicit) ecdsa_ptr: SignatureBuiltin*
+```
+
+#### `execute`
+
+This is the main an sole entrypoint to interact with the Account contract. It:
+
+1. Confirms that the Account has been initialized
+2. Takes the input and builds a [Message](#message-format) with it
+3. Validates the transaction signature
+4. Increments the nonce
+5. Calls the target contract with the intended function selector and calldata parameters
+6. Forwards the contract call response data as return value
+
+##### Parameters:
+```
+to: felt
+selector: felt
+calldata_len: felt
+calldata: felt*
+signature_len: felt
+signature: felt*
+```
+
+> Notice that the current signature scheme expects a 2-element array like `[sig_r, sig_s]`.
+
+
+##### Returns:
+```
+(implicit) syscall_ptr : felt*
+(implicit) pedersen_ptr: HashBuiltin*
+(implicit) range_check_ptr
+response: felt
+```
+
+## Extending the Account contract
+
+There's no clear contract extensibility pattern for Cairo smart contracts yet. In the meantime the best way to extend our contracts is copypasting and modifying them at your own risk. Since `execute` relies on it, we suggest changing how `is_valid_signature` works to explore different signature validation schemes such as multisig, or some guardian logic like in [Argent's account](https://github.com/argentlabs/argent-contracts-starknet/blob/de5654555309fa76160ba3d7393d32d2b12e7349/contracts/ArgentAccount.cairo).
+
+
+## L1 scape hatch mechanism
+
+*[unknown, to be defined]*
+
+
+## Paying for gas
+
+*[unknown, to be defined]*
+
+
+## Future work
```

Comment:
I honestly wouldn't include this list here. It will get out of date. The issue tracker is a better place for the information.